### ISSUE TITLE
Provide Python wheels

### ISSUE THREAD

**Issue by BSchilperoort** [csdms/bmi-python#60]
I tried using bmipy for a teachbooks BMI "toybox". However, micropip can only install packages with pure python wheels, but [only the source tarball is available](https://pypi.org/project/bmipy/2.0.1/#files) for `bmipy`.

Apart from enabling easy installation in WASM Python, having wheels available has more benefits:

1. Faster installation for pure Python packages.
1. Avoids arbitrary code execution for installation. (Avoids setup.py)
1. Allows better caching for testing and continuous integration.
1. Creates .pyc files as part of installation to ensure they match the Python interpreter used.
1. More consistent installs across platforms and machines.

(source: https://pythonwheels.com/)

<hr>

To create both a tarball and wheel, the command following command suffices:
```sh
> python -m build 
...
Successfully built bmipy-2.0.2.dev0.tar.gz and bmipy-2.0.2.dev0-py3-none-any.whl
```

**Comment by BSchilperoort** [comment]
Note that this should not require a new release: It is possible to publish new artifacts that are more specific for a given release.

**Comment by mdpiper** [comment]
@mcflugen has set up a [job](https://github.com/landlab/landlab/blob/master/.github/workflows/test.yml#L101) in the Landlab CI to build wheels. Perhaps this could be adapted for use here?

**Comment by mcflugen** [comment]
Yup, this would be easy to do. In fact, we already have a *nox* session that builds wheels (`nox -s build`).

**Comment by mcflugen** [comment]
@BSchilperoort I just uploaded a wheel for *v2.0.1*. Let me know if that works for you or if there is a problem.

Before the next release I'll add a job to our gha workflow to upload a source distribution and a wheel to PyPI on a release tag.

**Comment by BSchilperoort** [comment]
I went to https://jupyter.org/try-jupyter/lab/

and was able to install bmipy:
```py
import micropip
await micropip.install("bmipy")
```

And was able to paste in the BMI heat model code and run it successfully.
